storage/src/main/java/io/delta/storage/internal/ThreadUtils.java

```diff
@@ -0,0 +1,122 @@
+                    resultHolder.add(body.call());
+                } catch (Exception ex) {
+                    exceptionHolder.add(ex);
+                }
+            }
+        };
+        thread.setDaemon(isDaemon);
+        thread.start();
+        try {
+            thread.join();
+        } catch (InterruptedException e) {
+            throw new InterruptedIOException(e.getMessage());
+        }
+        if (!exceptionHolder.isEmpty()) {
+            Exception realException = exceptionHolder.get(0);
+            // Remove the part of the stack that shows method calls into this helper method
+            // This means drop everything from the top until the stack element
+            // ThreadUtils.runInNewThread(), and then drop that as well (hence the + 1 to start index).
+            List<StackTraceElement> currentThreadStackTrace =
+                    Arrays.asList(Thread.currentThread().getStackTrace());
+            //index of the stack element pointing to ThreadUtils.runInNewThread()
+            int startIndex = currentThreadStackTrace.stream()
+                    .filter(e -> e.getClassName().contains(ThreadUtils.class.getSimpleName()))
+                    .map(e -> currentThreadStackTrace.indexOf(e))
+                    .findFirst()
+                    .orElse(-1);
+            int endIndex = currentThreadStackTrace.size();
+            List<StackTraceElement> baseStackTrace;
+            if (startIndex == -1 || startIndex >= endIndex) {
+                baseStackTrace = Collections.emptyList();
+            } else {
+                //startIndex + 1 to drop the stack element ThreadUtils.runInNewThread()
+                baseStackTrace =
+                        currentThreadStackTrace.subList(startIndex + 1, currentThreadStackTrace.size());
+            }
+            // Remove the part of the new thread stack that shows methods call from this helper method
+            // This means take everything from the top until the stack element
+            StackTraceElement[] realExceptionStackTrace = realException.getStackTrace();
+            List<StackTraceElement> extraStackTrace = new ArrayList<>();
+            for (StackTraceElement st : realExceptionStackTrace) {
+                if (!st.getClassName().contains(ThreadUtils.class.getSimpleName())) {
+                    extraStackTrace.add(st);
+                }
+            }
+            // Combine the two stack traces, with a place holder just specifying that there
+            // was a helper method used, without any further details of the helper
+            List<StackTraceElement> placeHolderStackElem = Arrays.asList(
+                    new StackTraceElement(
+                            String.format(
+                                    "... run in separate thread using %s",
+                                    ThreadUtils.class.getSimpleName(),
+                                    " static method runInNewThread"), //Providing the helper class info.
```

Comment:
shouldn't this `String.format()` only take 2 args? "format" and then the input for the "%s"?